### ISSUE TITLE
Power Delivery Direction Reversing When Charging Off of Power Bank

### ISSUE THREAD

**Issue by Nicknakin** [linux-surface/linux-surface#944]
My Surface 3 running Fedora 36 will charge up to ~7% off of USB-C from a portable battery before deciding to instead charge the battery itself. I attempted to explicitly set the direction for power delivery but the /sys/class/typec device doesn't seem to exist.

This is the battery in question, it is capable of delivering enough power to charge the laptop, and it is definitely capable of negotiating with a Linux machine as it works fine with my Steam Deck.
https://www.amazon.com/dp/B08THFDRSZ?ref=ppx_yo2ov_dt_b_product_details&th=1

### Environment

- Hardware model: Surface Laptop 3
- Kernel version: Linux nicksurface 5.19.8-1.surface.fc36.x86_64 #1 SMP PREEMPT_DYNAMIC Wed Sep 14 23:21:37 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
- Distribution: Fedora Linux 36 (Workstation Edition) x86_64

<details><summary>`dmesg` output</summary>
```
Will provide next time battery is low enough
```

</details>

**Comment by qzed** [comment]
We might be missing a driver for that. On some devices, there's some HID Type-C switch driver (unfortunately I don't recall the exact name but you might be able to find it via the device manager in Windows). I believe us missing that driver could cause that, especially as the battery is managed by the embedded controller.

**Comment by Nicknakin** [comment]
Ah, I don't have windows on the device at all so I unfortunately couldn't check to get you more information. Is this something that's within scope of this project?

**Comment by qzed** [comment]
> Is this something that's within scope of this project?

Yes, definitely. Charging is handled by the EC, so it might be possible that we're missing some driver for that that connects the USB-C to the EC.

**Comment by NP-chaonay** [comment]
Idk if this useful information

Some time I can produce SLS to charge to powerbank but it is kind of rare, and can be fix just replug-in

I think this is controlled by EC and PD firmware, not driver or OS. This is what I guess. SLS Ubuntu Linux Surface don't have sys class type c file btw
